src/main/java/frc/robot/subsystems/vision/provider/CachingVision.java

```diff
@@ -0,0 +1,32 @@
+package frc.robot.subsystems.vision.provider;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Vision decorator with caching functionality (fixme requires implementation)
+ */
+public class CachingVision implements Vision{
```

Comment:
this isn't caching yet, right? I'm not missing something?

Reply:
Nope. Thats what the fixme is reminding me to do.